benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/tasks/running/CheckTrialResultTask.java

```diff
@@ -30,35 +31,38 @@ public CheckTrialResultTask(String trialID) {
   }
 
   @Override
-  public TrialResult call() throws Exception {
+  public TrialExecutionStatus call() throws Exception {
 
     logger.info("running - " + trialID);
 
     try {
 
-      RunStatus.Code trialStatus = trialModelDAO.getTrialStatus(trialID);
+      TrialStatus trialStatus = trialModelDAO.getTrialStatus(trialID);
 
       int retries = trialModelDAO.getNumRetries(trialID);
 
       String experimentID = BenchFlowConstants.getExperimentIDFromTrialID(trialID);
 
       int maxRetries = experimentModelDAO.getNumTrialRetries(experimentID);
 
-      TrialResult trialResult = null;
+      FailureStatus failureStatus = null;
 
       switch (trialStatus) {
-        case COMPLETED:
-          trialResult = TrialResult.SUCCESS;
-          break;
+
+        // TODO - add other experiment failures
+
+        case SUCCESS:
+          // not a failure
+          return TrialExecutionStatus.SUCCESS;
 
         case FAILED:
-          trialResult = TrialResult.FAILURE;
+          failureStatus = FailureStatus.EXECUTION;
           break;
 
-        case KILLED:
-        case KILLING:
-        case DENIED:
-          trialResult = TrialResult.FAILURE;
+        case RANDOM_FAILURE:
```

Comment:
Why don't you have an intermediate status also for this cases, so that you can decide all out of the switch statement?